Sources/eppo/RuleEvaluator.swift

```diff
@@ -48,57 +49,75 @@ public class RuleEvaluator {
         _ condition: TargetingCondition
     ) throws -> Bool
     {
-        if let value = subjectAttributes[condition.attribute] {
-            do {
-                switch condition.targetingOperator {
-                    case .GreaterThanEqualTo:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a >= b }
-                        );
-                    case .GreaterThan:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a > b }
-                        )
-                    case .LessThanEqualTo:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a <= b }
-                        )
-                    case .LessThan:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a < b }
-                        )
-                    case .Matches:
-                        return try Compare.compareRegex(
-                            value.stringValue(),
-                            condition.value.stringValue()
-                        )
-                    case .OneOf:
-                        return try Compare.isOneOf(
-                            value.stringValue(),
-                            condition.value.arrayValue()
-                        )
-                    case .NotOneOf:
-                        return try !Compare.isOneOf(
-                            value.stringValue(),
-                            condition.value.arrayValue()
-                        )
-                    default:
-                        throw Errors.UnexpectedValue
+        guard let value = subjectAttributes[condition.attribute] else {
+           return false
+        }
+        
+        do {
+            let comparisonResult: Bool
+            switch condition.targetingOperator {
+            case .GreaterThanEqualTo, .GreaterThan, .LessThanEqualTo, .LessThan:
```

Comment:
I really like this programming style! Makes a lot of sense to me to branch at this top level as the behavior is quite different from `matches` and `oneof/notoneof`

Reply:
💯